aten/src/ATen/native/native_functions.yaml

```diff
@@ -2571,6 +2571,11 @@
 
 - func: layer_norm(Tensor input, int[] normalized_shape, Tensor? weight=None, Tensor? bias=None, float eps=1e-05, bool cudnn_enable=True) -> Tensor
 
+- func: layer_norm.out(Tensor input, int[] normalized_shape, Tensor? weight=None, Tensor? bias=None, float eps=1e-05, bool cudnn_enable=True, *, Tensor(a!) out) -> Tensor(a!)
+  variants: function
+  dispatch:
+    CompositeExplicitAutograd: layer_norm_out
```

Comment:
out variants don't support autograd. So no need for dispatch key. And you could let the usual c++ overload resolution find your function?